airbyte-cdk/java/airbyte-cdk/core/src/test/java/io/airbyte/cdk/db/factory/DataSourceFactoryTest.java

```diff
@@ -93,6 +94,22 @@ void testCreatingMySQLDataSourceWithConnectionTimeoutSetBelowDefault() {
     }
   }
 
+  @Test
```

Comment:
Is this the only reason why we include mssqltestcontainer above? If so, we'd probably be better off making this class abstract, and asking individual source implementation to have their own implementation of this particular class.

Reply:
I agree in principle. I wonder if this extends to more than just the connectTimeout in the hikari pool as well. We can revisit in a separate PR.